### PR TITLE
fix some code style and missing phpDoc

### PR DESCRIPTION
**Pull request by tveronesi**
pulito codice e sistemato phpdoc mancanti o incompleti

### CODE REVIEW

File: src/CodiceFiscale/AbstractCalculator.php [hunk 1]
```diff
@@ -140,14 +140,14 @@ abstract class AbstractCalculator
...
      * Calculate the sum by the given dictionary for the given temporary codice fiscale.
      *
-     * @param $temporaryCodiceFiscale The temporary codice fiscale.
-     * @param $dictionaryArray The dictionary array.
-     * @param $i The start index value.
-     * @returns Returns the sum by the given dictionary for the given temporary codice fiscale.
+     * @param string $temporaryCodiceFiscale  The temporary codice fiscale.
+     * @param array $dictionaryArray  The dictionary array
+     * @param int $i  The start index value.
+     * @return int
```
Comment (DavidePastore) [inline src/CodiceFiscale/AbstractCalculator.php:150]: Here the `@return` description is missing. Is it wanted?

File: src/CodiceFiscale/AbstractCalculator.php [hunk 2]
```diff
@@ -140,14 +140,14 @@ abstract class AbstractCalculator
...
-    
+
     /**
      * Calculate the sum by the given dictionary for the given temporary codice fiscale.
      *
-     * @param $temporaryCodiceFiscale The temporary codice fiscale.
-     * @param $dictionaryArray The dictionary array.
-     * @param $i The start index value.
-     * @returns Returns the sum by the given dictionary for the given temporary codice fiscale.
+     * @param string $temporaryCodiceFiscale  The temporary codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/AbstractCalculator.php]: Are all spaces after the `@param` names duplicated for a particular reason?

File: src/CodiceFiscale/AbstractCalculator.php [hunk 3]
```diff
@@ -159,12 +159,12 @@ protected function calculateSumByDictionary($temporaryCodiceFiscale, $dictionary
...
     }
-    
+
     /**
      * Calculate the check digit.
      *
-     * @param $temporaryCodiceFiscale The first part of the codice fiscale.
-     * @returns Returns the check digit part of the codice fiscale.
+     * @param string $temporaryCodiceFiscale  The first part of the codice fiscale.
+     * @return string the check digit part of the codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/AbstractCalculator.php]: Can you please use an uppercase **T** for the initial **the**?

File: src/CodiceFiscale/Calculator.php [hunk 4]
```diff
@@ -35,7 +37,7 @@ public function __construct(Subject $subject, $properties = array())
     /**
      * Calculate the code fiscale.
      *
-     * @returns Returns the complete codice fiscale.
+     * @returns string the complete codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: Can you please use an uppercase **T** for the initial **the**?

File: src/CodiceFiscale/Calculator.php [hunk 5]
```diff
@@ -49,7 +51,7 @@ public function calculate()
     /**
      * Calculate all possibilities for the code fiscale.
      *
-     * @returns Returns the complete codice fiscale.
+     * @return array Returns the complete codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 6]
```diff
@@ -65,7 +67,7 @@ public function calculateAllPossibilities()
     /**
      * Calculate the surname part of the codice fiscale.
      *
-     * @returns Returns the surname part of the codice fiscale.
+     * @return string Returns the surname part of the codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 7]
```diff
@@ -83,7 +85,7 @@ private function calculateSurname()
     /**
      * Calculate the name part of the codice fiscale.
      *
-     * @returns Returns the name part of the codice fiscale.
+     * @return string Returns the name part of the codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 8]
```diff
@@ -103,9 +105,9 @@ private function calculateName()
     /**
      * Calculate small string for the given parameters (used by name and surname).
      *
-     * @param $consonants A consonants string.
-     * @param $string The small string.
-     * @returns Returns the calculated result for the small string.
+     * @param string $consonants   A consonants string.
+     * @param string $string  The small string.
+     * @return string Returns the calculated result for the small string.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 9]
```diff
@@ -119,7 +121,7 @@ private function calculateSmallString($consonants, $string)
     /**
      * Calculate the birth date and the gender.
      *
-     * @returns Returns the birth date and gender part of the codice fiscale.
+     * @returns string Returns the birth date and gender part of the codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 10]
```diff
@@ -136,7 +138,7 @@ private function calculateBirthDateAndGender()
     /**
      * Calculate the Belfiore code.
      *
-     * @returns Returns the Belfiore code.
+     * @return string Returns the Belfiore code.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 11]
```diff
@@ -169,12 +171,12 @@ private function calculateOmocodia($temporaryCodiceFiscale)
...
-     * @param $startingIndex The starting index.
-     * @param $endingIndex The ending index.
-     * @param $characterIndex The index to use to make the substitutions on the $temporaryCodiceFiscale.
-     * @returns Returns the temporary codice fiscale with the substitutions made.
+     * @param $divider int The divider.
+     * @param $temporaryCodiceFiscale string The first part of the codice fiscale on which make the substitutions.
+     * @param $startingIndex int The starting index.
+     * @param $endingIndex int The ending index.
+     * @param $characterIndex int The index to use to make the substitutions on the $temporaryCodiceFiscale.
+     * @return string Returns the temporary codice fiscale with the substitutions made.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Calculator.php [hunk 12]
```diff
@@ -146,8 +148,8 @@ private function calculateBelfioreCode()
     /**
      * Calculate the omocodia case (additional translation).
      *
-     * @param $temporaryCodiceFiscale The first part of the codice fiscale.
-     * @returns Returns the new codice fiscale.
+     * @param $temporaryCodiceFiscale string The first part of the codice fiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php:151]: `string` should be before `$temporaryCodiceFiscale` here.

File: src/CodiceFiscale/Calculator.php [hunk 13]
```diff
@@ -169,12 +171,12 @@ private function calculateOmocodia($temporaryCodiceFiscale)
...
-     * @param $temporaryCodiceFiscale The first part of the codice fiscale on which make the substitutions.
-     * @param $startingIndex The starting index.
-     * @param $endingIndex The ending index.
-     * @param $characterIndex The index to use to make the substitutions on the $temporaryCodiceFiscale.
-     * @returns Returns the temporary codice fiscale with the substitutions made.
+     * @param $divider int The divider.
+     * @param $temporaryCodiceFiscale string The first part of the codice fiscale on which make the substitutions.
+     * @param $startingIndex int The starting index.
+     * @param $endingIndex int The ending index.
+     * @param $characterIndex int The index to use to make the substitutions on the $temporaryCodiceFiscale.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php:178]: All these parameters should have the type before the name of the parameters.

File: src/CodiceFiscale/Calculator.php [hunk 14]
```diff
@@ -187,13 +189,11 @@ private function replaceOmocodiaSection($divider, $temporaryCodiceFiscale, $star
     }
 
     /**
-     * Clean the string removing some characters.
-     *
-     * @param $string The string to clean.
-     * @returns Returns a clean string.
+     * @param $string string The string to clean.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Calculator.php:192]: `string` should be before `$string` here.

File: src/CodiceFiscale/Checker.php [hunk 15]
```diff
@@ -19,10 +19,10 @@ class Checker
     const ALL_OMOCODIA_LEVELS = -1;
 
     /**
-     * Create a Codice Fiscale instance.
+     * Checker constructor.
      *
-     * @param Subject $subject The subject.
-     * @param $properties Array of additional properties.
+     * @param Subject $subject The subject
```
Comment (DavidePastore) [inline src/CodiceFiscale/Checker.php]: You can leave the final dot here.

File: src/CodiceFiscale/Checker.php [hunk 16]
```diff
@@ -19,10 +19,10 @@ class Checker
...
+     * @param array $properties additional properties.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Checker.php]: The **a** of **additional** should be uppercase.

File: src/CodiceFiscale/Subject.php [hunk 17]
```diff
@@ -53,7 +58,7 @@ public function __construct($properties)
     /**
      * Get the name.
      *
-     * @return Returns the name.
+     * @return string Returns the name.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Subject.php]: You could remove Returns here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Subject.php [hunk 18]
```diff
@@ -71,7 +76,7 @@ public function setName($name)
...
```
Comment (DavidePastore) [inline src/CodiceFiscale/Subject.php]: You could remove Returns here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Subject.php [hunk 19]
```diff
@@ -89,7 +94,7 @@ public function setSurname($surname)
     /**
      * Get the birthDate.
      *
-     * @return Returns the birthDate.
+     * @return DateTime Returns the birthDate.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Subject.php]: You could remove Returns here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Subject.php [hunk 20]
```diff
@@ -111,7 +118,7 @@ public function setBirthDate($birthDate)
     /**
      * Get the gender.
      *
-     * @return Returns the gender.
+     * @return string Returns the gender.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Subject.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

File: src/CodiceFiscale/Subject.php [hunk 21]
```diff
@@ -129,7 +136,7 @@ public function setGender($gender)
     /**
      * Get the belfioreCode.
      *
-     * @return Returns the belfioreCode.
+     * @return string Returns the belfioreCode.
```
Comment (DavidePastore) [inline src/CodiceFiscale/Subject.php]: You could remove **Returns** here and use an uppercase **T** for the initial **the**.

**Review by DavidePastore — CHANGES_REQUESTED** [review]
Thanks for your work on this! You can see there a lot of review comments but most of them are replicated over the PR. When you'll make these changes I'll merge it. 😄

File: src/CodiceFiscale/AbstractCalculator.php (hunk 1 above)
Comment (tveronesi) [reply src/CodiceFiscale/AbstractCalculator.php:150]: return info/description is implicit by method name

**Comment by tveronesi** [comment]
ok

**Comment by tveronesi** [comment]
reviewed all requests!

**Review by DavidePastore — APPROVED** [review]
Thanks for your changes here.

**Pull request merged**